From 3b5ef79b57dd3d0fa45b74dad6a7d62d85c205ae Mon Sep 17 00:00:00 2001
From: Ian Denhardt <ian@zenhack.net>
Date: Wed, 6 Oct 2021 20:41:44 -0400
Subject: [PATCH 2/4] Redirect / to a single unique document.

---
 lib/response.js | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/lib/response.js b/lib/response.js
index 853f80a4e..b0efa4377 100644
--- a/lib/response.js
+++ b/lib/response.js
@@ -22,6 +22,20 @@ function showIndex (req, res, next) {
   const authStatus = req.isAuthenticated()
   const deleteToken = ''
 
+  // Sandstorm edit: redirect to a single note. Create a new
+  // one if one does not already exist.
+  models.Note.findOne({}).then((note) => {
+    let loc = '/new';
+    if(note) {
+      console.log(note)
+      loc = note.dataValues.shortid;
+    }
+    console.log("Location: ", loc);
+    res.writeHead(301, {'Location': loc})
+    res.end();
+  });
+  return;
+
   const data = {
     signin: authStatus,
     infoMessage: req.flash('info'),
-- 
2.39.2

